carbonmark-api/src/routes/users/create.ts

```diff
@@ -72,21 +85,22 @@ const handler = (fastify: FastifyInstance) =>
     if (user.exists) {
       return reply.code(403).send({
         code: 403,
-        error: "This user is already registered!",
+        error: "This wallet address is already registered!",
       });
     }
 
+    // Check if the handle already exists in our database
     const usersRef = fastify.firebase.firestore().collection("users");
 
     const userSnapshot = await usersRef
       .where("handle", "==", handle.toLowerCase())
       .limit(1)
       .get();
-    // If no documents are found, return a 404 error
+
     if (!userSnapshot.empty) {
       return reply.code(403).send({
         code: 403,
-        error: "This user is already registered!",
+        error: "A user with this handle is already registered!",
       });
     }
```

Comment:
I find these checks a little confusing:

* `user.exists == true` => **wallet** already registered
* `!userSnapshot.empty == true` => **user** with this handle is already registered

Especially since the former checks explicitly if `user.exists`…

Reply:
I can see why, in the first case we're confirming that no Document in firebase exists with that wallet (this is the primary key).

In the second we want to make sure that no one is creating a new account with a new address (primary key) but using a handle that someone else has already set on their Document.

Does that make sense?